### PR TITLE
module articles category - add random order option

### PR DESCRIPTION
**Pull request by alikon**
follow up #6648
add random order option  to articles category module
##### How to test

apply the patch
select the random option on `Article Field to Order By`
modules-articles-category
#### Expected result

the articles are showed randomly orderdered
#### Additional comment

should work on all supported db's

### CODE REVIEW

File: modules/mod_articles_category/helper.php [hunk 1]
```diff
@@ -163,8 +163,16 @@ public static function getList(&$params)
 		}
 
 		// Ordering
-		$articles->setState('list.ordering', $params->get('article_ordering', 'a.ordering'));
-		$articles->setState('list.direction', $params->get('article_ordering_direction', 'ASC'));
+		$ordering = $params->get('article_ordering', 'a.ordering');
+		if (trim($ordering) == 'rand()')
```
Comment (zero-24) [inline modules/mod_articles_category/helper.php]: can we add a clean line bevor the `if`? I think as we don't require `rand()` we should use `random` as option value as i think it is better readable what do you think?

File: modules/mod_articles_category/mod_articles_category.xml [hunk 2]
```diff
@@ -270,6 +270,7 @@
 					<option value="modified">MOD_ARTICLES_CATEGORY_OPTION_MODIFIED_VALUE</option>
 					<option value="publish_up">MOD_ARTICLES_CATEGORY_OPTION_STARTPUBLISHING_VALUE</option>
 					<option value="a.publish_down">MOD_ARTICLES_CATEGORY_OPTION_FINISHPUBLISHING_VALUE</option>
+					<option value="rand()">MOD_ARTICLES_CATEGORY_OPTION_RANDOM_VALUE</option>
```
Comment (zero-24) [inline modules/mod_articles_category/mod_articles_category.xml]: I think as we don't require `rand()` we should use `random` as option value as i think it is better readable what do you think?

**Comment by zero-24** [comment]
I have just comment on two issues. I can test it later today ;)

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on eb2ddf1f76c77e1395947800cd1204f8bc52e855<br /><br />It works. Lets work out the two inline comments ;)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8538">issues.joomla.org/joomla-cms/8538</a>.</sub>

File: modules/mod_articles_category/mod_articles_category.xml (hunk 2 above)
Comment (Fedik) [inline modules/mod_articles_category/mod_articles_category.xml]: I agree with @zero-24

**Comment by alikon** [comment]
comments should be ok now ;)

**Comment by zero-24** [comment]
Great :+1:

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 72d60cefcadeaabb111f6428670d5e28062258b1<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8538">issues.joomla.org/joomla-cms/8538</a>.</sub>

**Comment by brianteeman** [comment]
Thanks for testing - setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8538">issues.joomla.org/joomla-cms/8538</a>.</sub>

**Comment by pjwiseman** [comment]
I have tested this item :white_check_mark: successfully on 72d60cefcadeaabb111f6428670d5e28062258b1<br /><br />Piggy-backed of zero's test with a code review of some minor changes. Looks good.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8538">issues.joomla.org/joomla-cms/8538</a>.</sub>

**Pull request closed without merging**